graph/impl/src/main/java/org/jboss/windup/graph/SetInPropertiesHandler.java

```diff
@@ -0,0 +1,189 @@
+import com.tinkerpop.frames.modules.MethodHandler;
+
+/**
+ * Handles @SetInProperties Set<String,String>.
+ *
+ * @author <a href="mailto:ozizka@redhat.com">Ondrej Zizka</a>
+ */
+public class SetInPropertiesHandler implements MethodHandler<SetInProperties>
+{
+    private static final Logger log = Logging.get(SetInPropertiesHandler.class);
+
+    private static final String SET_VERTEX_PROP_VALUE = "1";
+
+
+    @Override
+    public Class<SetInProperties> getAnnotationType()
+    {
+        return SetInProperties.class;
+    }
+
+
+    /**
+     * The handling method.
+     */
+    @Override
+    public Object processElement(Object frame, Method method, Object[] args, SetInProperties ann,
+                FramedGraph<?> framedGraph, Element elm)
+    {
+        String methodName = method.getName();
+        if (methodName.startsWith("get"))
+            return handleGetter((Vertex) elm, method, args, ann, framedGraph);
+
+        else if (methodName.startsWith("set"))
+            handleSetter((Vertex) elm, method, args, ann, framedGraph);
+
+        else if (methodName.startsWith("addAll"))
+            handleAdder((Vertex) elm, method, args, ann, framedGraph);
+
+        else if (methodName.startsWith("add"))
+            handleAddAll((Vertex) elm, method, args, ann, framedGraph);
+
+        else
+            throw new WindupException("Only get*, set*, add*, and addAll* method names are supported for @"
+                    + SetInProperties.class.getSimpleName() + ", found at: " + method.getName());
+
+        return frame;
+    }
+
+    /**
+     * Getter
+     */
+    private Set<String> handleGetter(Vertex vertex, Method method, Object[] args, SetInProperties ann, FramedGraph<?> framedGraph)
+    {
+        if (args != null && args.length != 0)
+            throw new WindupException("Method must take zero arguments");
+
+        Set<String> set = new HashSet<>();
+        String prefix = preparePrefix(ann);
+
+        Set<String> keys = vertex.getPropertyKeys();
+        for (String key : keys)
+        {
+            String tail = key;
+            if (!prefix.isEmpty())
+            {
+                if(!key.startsWith(prefix))
+                    continue;
+                else
+                    tail = key.substring(prefix.length());
+            }
+
+            set.add(tail);
+        }
+
+        return set;
+    }
+
+    /**
+     * Setter
+     */
+    private void handleSetter(Vertex vertex, Method method, Object[] args, SetInProperties ann, FramedGraph<?> framedGraph)
+    {
+        // Argument.
+        if (args == null || args.length != 1)
+            throw new WindupException("Method must take one argument: " + method.getName());
+
+        if (!(args[0] instanceof Set))
+            throw new WindupException("Argument of " + method.getName() + " must be a Set<String>, but is: " + args[0].getClass());
+
+        @SuppressWarnings("unchecked")
+        Set<String> newSet = (Set<String>) args[0];
```

Comment:
Because we do not need anything else, FWICT. Plus, it is implemented to be stored as property keys.